moveit_setup_assistant/src/widgets/controllers_widget.cpp

```diff
@@ -0,0 +1,834 @@
+  connect(joints_widget_, SIGNAL(cancelEditing()), this, SLOT(cancelEditing()));
+  connect(joints_widget_, SIGNAL(doneEditing()), this, SLOT(saveJointsScreen()));
+  connect(joints_widget_, SIGNAL(previewSelected(std::vector<std::string>)), this,
+          SLOT(previewSelectedJoints(std::vector<std::string>)));
+
+  // Joints Groups Widget
+  joint_groups_widget_ = new DoubleListWidget(this, config_data_, "Group Joints Collection", "Group");
+  connect(joint_groups_widget_, SIGNAL(cancelEditing()), this, SLOT(cancelEditing()));
+  connect(joint_groups_widget_, SIGNAL(doneEditing()), this, SLOT(saveJointsGroupsScreen()));
+  connect(joint_groups_widget_, SIGNAL(previewSelected(std::vector<std::string>)), this,
+          SLOT(previewSelectedGroup(std::vector<std::string>)));
+
+  // Controller Edit Widget
+  controller_edit_widget_ = new ControllerEditWidget(this, config_data_);
+  connect(controller_edit_widget_, SIGNAL(cancelEditing()), this, SLOT(cancelEditing()));
+  connect(controller_edit_widget_, SIGNAL(deleteController()), this, SLOT(deleteController()));
+  connect(controller_edit_widget_, SIGNAL(save()), this, SLOT(saveControllerScreenEdit()));
+  connect(controller_edit_widget_, SIGNAL(saveJoints()), this, SLOT(saveControllerScreenJoints()));
+  connect(controller_edit_widget_, SIGNAL(saveJointsGroups()), this, SLOT(saveControllerScreenGroups()));
+
+  // Combine into stack
+  stacked_layout_ = new QStackedLayout(this);
+  stacked_layout_->addWidget(controllers_tree_widget);  // screen index 0
+  stacked_layout_->addWidget(joints_widget_);           // screen index 1
+  stacked_layout_->addWidget(controller_edit_widget_);  // screen index 2
+  stacked_layout_->addWidget(joint_groups_widget_);     // screen index 3
+
+  // Finish GUI -----------------------------------------------------------
+
+  // Create Widget wrapper for layout
+  QWidget* stacked_layout_widget = new QWidget(this);
+  stacked_layout_widget->setLayout(stacked_layout_);
+
+  layout->addWidget(stacked_layout_widget);
+
+  this->setLayout(layout);
+
+  // Create joint publisher -----------------------------------------
+  ros::NodeHandle nh;
+
+  // Create scene publisher for later use
+  pub_robot_state_ = nh.advertise<moveit_msgs::DisplayRobotState>(MOVEIT_ROBOT_STATE, 1);
+}
+
+// ******************************************************************************************
+// Create the main tree view widget
+// ******************************************************************************************
+QWidget* ControllersWidget::createContentsWidget()
+{
+  // Main widget
+  QWidget* content_widget = new QWidget(this);
+
+  // Basic widget container
+  QVBoxLayout* layout = new QVBoxLayout(this);
+
+  QHBoxLayout* upper_controls_layout = new QHBoxLayout();
+
+  // Add default controller
+  QPushButton* btn_add_default_ = new QPushButton("&Add Joints Follow Trajectory Controllers", this);
```

Comment:
_ remove

Reply:
Rename button: "Auto Add FollowJointTrajectory Controllers For Each Planning Group"